### ISSUE TITLE
[BUG]: Awful mode missing noodles effect when using the hardware renderers in PaRappa the Rapper 2

### ISSUE THREAD

**Issue by Roboguy420** [PCSX2/pcsx2#8390]
### Describe the Bug

When using a hardware renderer, on some of the stages in the game (stages 1, 3, 4, 5, and 6) the noodle effects around the border disappear completely when entering awful mode.

### Reproduction Steps

Choose any of the hardware renderers. Go to stage 1, 3, 4, 5, or 6. Perform poorly enough that you enter awful mode.

### Expected Behavior

There should be noodle effects around the border (See, for example, [IcedTeaGaming's video of PaRappa 2's stage 1 awful mode running on original PS2 hardware](https://youtu.be/QpSxjX6tcqQ))

### PCSX2 Revision

v1.7.4225

### Operating System

Windows 10 (64bit)

### CPU

i3-12100F

### GPU

RTX 3060

### GS Settings

Renderer is OpenGL (although the issue occurs when using any of the renderers apart from software)
VSync is enabled
Internal resolution is 3x native (issue also occurs on native)
Dithering is scaled (issue also occurs on unscaled)
Blending accuracy is full (issue also occurs on high and maximum)

### GS Window Screenshots

![PaRappa the Rapper 2_SCES-50408_20230313152059](https://user-images.githubusercontent.com/108060195/224746848-8b035673-9e56-47c7-8411-2c32343072ad.png)

### Logs & Dumps

https://drive.google.com/file/d/1d5GTojusCptevFloa1TwDJJIAyBCq_Br/view?usp=sharing

**Comment by refractionpcsx2** [comment]
Interesting, ticking "Disable depth emulation" in the hw fixes seems to bring it back.

Alright thanks.

**Comment by cuckydev** [comment]
> Interesting, ticking "Disable depth emulation" in the hw fixes seems to bring it back.
> 
> Alright thanks.

This is true. Unfortunately, disabling depth emulation introduces many other rendering issues. I haven't looked at the full GS dump, but the animated background is probably modifying the depth buffer (clearing) in a way that doesn't work on the hardware renderers.

Interestingly, the issue does not occur on Stage 7 of the game. For convenience, I have attached a GS dump.
[PaRappa the Rapper 2_SCPS-15017_20230516210046.zip](https://github.com/PCSX2/pcsx2/files/11493203/PaRappa.the.Rapper.2_SCPS-15017_20230516210046.zip)

**Comment by cuckydev** [comment]
As well, the title screen and opening sequence uses the same effect of the waving noodle background and the foreground circle of noodles.
[PaRappa the Rapper 2_SCPS-15017_20230516210531.zip](https://github.com/PCSX2/pcsx2/files/11493221/PaRappa.the.Rapper.2_SCPS-15017_20230516210531.zip)

I should add that no game properties other than Disable Depth Emulation address this issue. It is likely a core issue with the hardware renderers and how they handle depth commands.

**Comment by stenzek** [comment]
Game draws the noodle things to a 256x80 target at 0x1400 with a BW of 10 (who knows why..).
Then it draws something to depth at 0x1400 to the right of it.
It breaks because we don't transfer the valid area, and it doesn't download. But fixing that creates another issue - the round trip from C32->Z32->Download results in a loss of precision, and the effect has incorrect colours.

CPU sprite render BW of 10 works, but breaks other parts of the game as expected. Could CRC hack it, but that's pretty gross.

**Comment by cuckydev** [comment]
So this is a fundamental problem with the hardware renderers then? I wonder why this game in particular has so many weird things going on.

**Comment by refractionpcsx2** [comment]
It's not an unusual case :P You have to think, that GS memory is not linear, and PS2 developers had free reign to do whatever they liked with it, so they came up with tons of weird and wonderful ways to squeeze extra graphical finesse out of the PS2's limited hardware.

The problem we face is for the hardware renderer to be fast (and allow for upscaling) we have to try and reinterpret everything it does in a linear way.   The problem being is a bunch of things can't really be done, or they do things in such a way it looks like they're doing something else, and then that breaks another game which relies on similar behaviour, we spend a lot of time playing cat and mouse with the hardware renderers.

So stens suggestion of a CRC hack (because fixing that breaks other things, surprise lol) isn't uncommon to try and work around strange behaviours.

**Comment by LeWolfYT** [comment]
It’s amazing how many issues with PTR2 are fixed by disabling depth emulation.

**Comment by cuckydev** [comment]
> It’s amazing how many issues with PTR2 are fixed by disabling depth emulation.

with all the other stuff that's broken by it, i don't think it's worth it :P

**Comment by stenzek** [comment]
"Disabling depth emulation" is a terrible name for what it does, it just prevents downloads of depth targets, and them being used as sources. Depth buffers are still absolutely used for testing and writing.

Regardless, it's a complete hack and is not a solution, since for sources you're loading garbage from local memory instead of the actual target.
